package/MDAnalysis/coordinates/chain.py

```diff
@@ -56,13 +56,99 @@
 from . import core
 
 
+def multi_level_argsort(l):
+    """
+    This sorts based on the start time.
+
+    Parameters
+    ----------
+    l : list
+
+    Returns
+    -------
+    indices
+    """
+    return [el[0] for el in sorted(enumerate(l), key=lambda x: x[1][0])]
+
+
+def filter_times(times, dt):
+    """Given a list of start and end times this function filters out any duplicate
+    time steps preferring the last tuple.
+
+    ((0, 3), (0, 3)) -> (1, )
+    ((0, 3), (0, 4)) -> (1, )
+    ((0, 3), (3, 4)) -> (0, 1)
+    ((0, 3), (2, 5), (4, 9)) -> (1, 2, 3)
+
+
+    Parameters
+    ----------
+    times :
+        sorted list of times
+    dt :
+        timestep between two frames
+
+    Returns
+    -------
+    indices of times to used with overlaps removed
+
+    """
+    # Special cases
+    if len(times) == 1:
+        return [0, ]
+    elif len(times) == 2:
+        if times[0][0] < times[1][0]:
+            return [0, 1]
+        elif np.allclose(times[0][0], times[1][0]):
+            return [1, ]
+        else:
+            return [0, ]
+    if np.unique(times).size == 2:
+        return [len(times) - 1, ]
+
+    # more then 2 unique time entries
+
+    # special case first times are equal
+    if not np.allclose(times[0][0], times[1][0]):
+        used_idx = [0, ]
+        offset = 1
+    else:
+        used_idx = [1, ]
+        offset = 2
+
+    for i, (first, middle, last) in enumerate(zip(times[:-2], times[1:-1], times[2:])):
+        if not np.allclose(middle[1] - middle[0], dt):
+            if (middle[0] <= first[1]) and (last[0] <= middle[1]):
+                used_idx.append(i + offset)
+        elif (middle[0] <= first[1]):
+            used_idx.append(i + offset)
+
+    # take care of first special case
+    if (times[-2][1] <= times[-1][1]):
+        used_idx.append(len(times) - (2-offset))
+
+    return used_idx
+
+
 class ChainReader(base.ProtoReader):
     """Reader that concatenates multiple trajectories on the fly.
 
     The :class:`ChainReader` is used by MDAnalysis internally to
     represent multiple trajectories as one virtual trajectory. Users
     typically do not need to use the :class:`ChainReader` explicitly.
 
+    Chainreader can also handle a continuous trajectory split over several
+    files. To use this passe the 'continuous == True' keyword argument. An
+    example take the following trajectory that is split into three parts. The
+    colum represents the time and the trajectory segments overlap. With the
+    continuous chainreader only the frames marked with a + will be read.
+
+    part01:  ++++--
+    part02:      ++++++-
+    part03:            ++++++++
+
+    The default chainreader will read all frames.
+
```

Comment:
Add the limitation that this is currently only supported for Gromacs XDR trajectories (XTC/TRR).